TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -732,7 +732,7 @@ public static DataDefinition CheckVariable(Node node, StorageArea storageArea, b
                         //We allow modification on indices because semantically they are value types, the procedure uses a copy of the input (issue #1789).
                         //Also format 5 set statements are allowed because it does not affect the input for the caller (issue #1625).
                         //set (address of)? identifier(pointer) TO (address of)? identifier | NULL
-                        bool isModificationAllowed = dataDefinitionFound.IsIndex
+                        bool isModificationAllowed = dataDefinitionFound.IsTableIndex
```

Comment:
So what happens for an index that's not a table index (i.e an index variable) ?

Reply:
An index variable is represented as an instance of `DataDefinition` (not `IndexDefinition`) so `IsTableIndex` is `false`, its usage is `Index` and its `PhysicalLength` is 4.